swiftide/src/integrations/treesitter/splitter.rs

```diff
@@ -109,26 +109,46 @@ impl CodeSplitter {
     /// # Returns
     ///
     /// * `Vec<String>` - A vector of code chunks as strings.
-    fn chunk_node(&self, node: Node, source: &str, mut last_end: usize) -> Vec<String> {
+    fn chunk_node(
+        &self,
+        node: Node,
+        source: &str,
+        mut last_end: usize,
+        current_chunk: Option<String>,
+    ) -> Vec<String> {
         let mut new_chunks: Vec<String> = Vec::new();
-        let mut current_chunk = String::new();
+        let mut current_chunk = current_chunk.unwrap_or(String::new());
 
         for child in node.children(&mut node.walk()) {
-            if child.end_byte() - child.start_byte() > self.max_bytes() {
-                // Child is too big, recursively chunk the child
-                if !current_chunk.is_empty() && current_chunk.len() > self.min_bytes() {
-                    new_chunks.push(current_chunk);
+            assert!(
+                current_chunk.len() <= self.max_bytes(),
+                "Chunk too big: {} > {}",
+                current_chunk.len(),
+                self.max_bytes()
+            );
```

Comment:
How do assertions work outside of tests? Do they log if the condition passes when debug assertions are enabled?

Reply:
No that's `debug_assert`, regular `assert` always cause a panic, both in debug and release builds. It's a technique used in high reliability projects like databases and operating systems to ensure invariants hold. Of course in Rust there's no risk of memory corruption so `assert` might be less useful outside of `unsafe` blocks. I'm fine with removing them if you think it's unnecessary.

Reply:
I think you have a point that it's good to have a double check, it really shouldn't ever happen. Are you sure you want to panic? It would break any running pipeline. You could also debug assert or log an error, or both.